src/app/static/src/app/components/plots/choropleth/Choropleth.vue

```diff
@@ -5,27 +5,31 @@
                  :selectedFilterOptions="selections.selectedFilterOptions"
                  @update="onFilterSelectionsChange"></filters>
         <div id="chart" :class="includeFilters ? 'col-md-9' : 'col-md-12'">
-            <l-map ref="map" style="height: 800px; width: 100%" @ready="() => updateBounds()">
-                <template v-for="feature in currentFeatures" :key="feature.id">
+            <l-map ref="map" style="height: 800px; width: 100%" @ready="updateBounds" @vnode-updated="updateBounds">
+                <template v-for="feature in currentFeatures" :key="feature.properties.area_id">
                     <l-geo-json :geojson="feature"
                                 :options="options"
                                 :optionsStyle="{...style, fillColor: getColor(feature)}">
                     </l-geo-json>
                 </template>
                 <map-empty-feature v-if="emptyFeature"></map-empty-feature>
-                <reset-map v-else @reset-view="updateBounds"></reset-map>
-                <map-control :initialDetail=selections.detail
-                             :indicator=selections.indicatorId
-                             :show-indicators="true"
-                             :indicators-metadata="indicators"
-                             :level-labels="featureLevels"
-                             @detail-changed="onDetailChange"
-                             @indicator-changed="onIndicatorChange"></map-control>
-                <map-legend v-show="!emptyFeature"
-                            :metadata="colorIndicator"
-                            :colour-scale="indicatorColourScale"
-                            :colour-range="colourRange"
-                            @update="updateColourScale"></map-legend>
+                <template v-else>
+                    <reset-map @reset-view="updateBounds"></reset-map>
+                </template>
+                <template>
+                    <map-control :initialDetail="selections.detail"
+                                :indicator="selections.indicatorId"
+                                :show-indicators="true"
+                                :indicators-metadata="indicators"
+                                :level-labels="featureLevels"
+                                @detail-changed="onDetailChange"
+                                @indicator-changed="onIndicatorChange"></map-control>
+                    <map-legend v-show="!emptyFeature"
+                                :metadata="colorIndicator"
+                                :colour-scale="indicatorColourScale"
+                                :colour-range="colourRange"
+                                @update="updateColourScale"></map-legend>
+                </template>
```

Comment:
so I can remove the template nesting on the `reset-map` part but not this bigger chunk, if you change a filter so that `emptyFeature` is true you actually get an "insertBefore on Node" DOMException. No idea why!